cmd/api/src/queries/graph.go

```diff
@@ -383,6 +384,11 @@ func (s *GraphQuery) prepareGraphQuery(rawCypher string, disableCypherQC bool) (
 	} else if complexityMeasure, err := analyzer.QueryComplexity(queryModel); err != nil {
 		return graphQuery, newQueryError(err)
 	} else if !disableCypherQC && complexityMeasure.Weight > MaxQueryComplexityWeightAllowed {
+		// log query details if it is rejected due to high complexity
+		timeoutEvent := log.WithLevel(log.LevelError)
+		timeoutEvent.Str("query", graphQuery.strippedQuery)
+		timeoutEvent.Msg(fmt.Sprintf("Query rejected. Query weight: %.2f. Maximum allowed weight: %d", graphQuery.complexity.Weight, MaxQueryComplexityWeightAllowed))
+
```

Comment:
nit: timeoutEvent makes me think a timeout happened but this is bailing before actually executing the query. 

It might be helpful to add the constituent complexity values to the logged event so the user knows what levers they can pull to reduce complexity like number of patterns or projections.